### PR TITLE
[FLINK-7477] [FLINK-7480] Various improvements to Flink scripts

### PR DESCRIPTION
**Pull request by aljoscha**

### CODE REVIEW

File: flink-dist/src/main/flink-bin/bin/config.sh [hunk 1]
```diff
@@ -351,8 +351,20 @@ if [ -z "$HADOOP_CONF_DIR" ]; then
     fi
 fi
 
+# try and set HADOOP_CONF_DIR to some common default if it's not set
+if [ -z "$HADOOP_CONF_DIR" ]; then
+    if [ -d "/etc/hadoop/conf" ]; then
+        HADOOP_CONF_DIR="/etc/hadoop/conf"
```
Comment (rmetzger) [inline flink-dist/src/main/flink-bin/bin/config.sh]: I would suggest to print a message to the user that we are using this HADOOP_CONF_DIR

File: flink-dist/src/main/flink-bin/bin/config.sh [hunk 2]
```diff
@@ -351,8 +351,20 @@ if [ -z "$HADOOP_CONF_DIR" ]; then
...
+    fi
+fi
+
 INTERNAL_HADOOP_CLASSPATHS="${HADOOP_CLASSPATH}:${HADOOP_CONF_DIR}:${YARN_CONF_DIR}"
 
+# check if the "hadoop" binary is available, if yes, use that to augment the CLASSPATH
+if command -v hadoop >/dev/null 2>&1; then
+    INTERNAL_HADOOP_CLASSPATHS="${HADOOP_CLASSPATH}:${HADOOP_CONF_DIR}:${YARN_CONF_DIR}:`hadoop classpath`"
```
Comment (rmetzger) [inline flink-dist/src/main/flink-bin/bin/config.sh]: I would actually append `INTERNAL_HADOOP_CLASSPATHS` instead of overwriting it.

**Comment by aljoscha** [comment]
@rmetzger I addressed your comments.

**Comment by rmetzger** [comment]
+1 to merge (assuming `command` is available on all operating systems / or is a bash command?)

**Comment by aljoscha** [comment]
It's a bash command, AFAIK.

**Comment by ruankd** [comment]
Hey, notice that this [commit](https://github.com/apache/flink/commit/0a0f6ed6c3d6cff702e4322293340274bea5e7d9) is part of this PR but it not merged into branch 1.5 and 1.6, neither in master. I wonder whether it will be merged?

**Comment by lfrancke** [comment]
I just stumbled across this as well.

The commit @ruankd referred to was indeed not committed.

**Pull request closed without merging**